changes/mapper.go

```diff
@@ -46,6 +46,9 @@ func NewTagMapper(tags ...string) *TagMapper {
 // KeyIndexes implements the KeyMapper interface and returns the keys and their
 // locations in the value's type.
 func (mapper *TagMapper) KeyIndexes(value reflect.Value) (KeyIndexes, error) {
+	mapper.RLock()
+	defer mapper.RUnlock()
```

Comment:
Hrm, I think you'll need to be explicit rather than use defer (my bad) because otherwise you'll hit a deadlock when you try and `mapper.Lock()` inside `mapper.registerValue(value)`